app/assets/javascripts/angular/controllers/dashboard_ctrl.js

```diff
@@ -30,6 +30,31 @@ angular.module("Prometheus.controllers")
     $scope.$broadcast('redrawGraphs');
   }
 
+  $scope.generateStaticDashboard = function(event) {
+    // TODO: When the input with the static dashboard link is visible,
+    // should clicking the "generate" button close the input, or
+    // generate a NEW static link?
```

Comment:
I think closing would be both the better and more consistent thing... haven't read the rest of the code yet, but one question would be whether we can detect if anything has changed and only regenerate a link in that case (as we do with individual graph permalinks, I think)